### PR TITLE
Fix #1677, Add Message API Functional Test

### LINKED ISSUE #1677: Add functional tests for cFE Message header APIs
**Is your feature request related to a problem? Please describe.**
Need open source functional tests for certifiability

**Describe the solution you'd like**
Add functional tests for cFE Message header APIs
    CFE_MSG_Init - Initialize a message.
    CFE_MSG_GetSize - Gets the total size of a message.
    CFE_MSG_SetSize - Sets the total size of a message.
    CFE_MSG_GetType - Gets the message type.
    CFE_MSG_SetType - Sets the message type.
    CFE_MSG_GetHeaderVersion - Gets the message header version.
    CFE_MSG_SetHeaderVersion - Sets the message header version.
    CFE_MSG_GetHasSecondaryHeader - Gets the message secondary header boolean.
    CFE_MSG_SetHasSecondaryHeader - Sets the message secondary header boolean.
    CFE_MSG_GetApId - Gets the message application ID.
    CFE_MSG_SetApId - Sets the message application ID.
    CFE_MSG_GetSegmentationFlag - Gets the message segmentation flag.
    CFE_MSG_SetSegmentationFlag - Sets the message segmentation flag.
    CFE_MSG_GetSequenceCount - Gets the message sequence count.
    CFE_MSG_SetSequenceCount - Sets the message sequence count.
    CFE_MSG_GetNextSequenceCount - Gets the next sequence count value (rolls over if appropriate)
    CFE_MSG_GetEDSVersion - Gets the message EDS version.
    CFE_MSG_SetEDSVersion - Sets the message EDS version.
    CFE_MSG_GetEndian - Gets the message endian.
    CFE_MSG_SetEndian - Sets the message endian.
    CFE_MSG_GetPlaybackFlag - Gets the message playback flag.
    CFE_MSG_SetPlaybackFlag - Sets the message playback flag.
    CFE_MSG_GetSubsystem - Gets the message subsystem.
    CFE_MSG_SetSubsystem - Sets the message subsystem.
    CFE_MSG_GetSystem - Gets the message system.
    CFE_MSG_SetSystem - Sets the message system.
    CFE_MSG_GenerateChecksum - Calculates and sets the checksum of a message.
    CFE_MSG_ValidateChecksum - Validates the checksum of a message.
    CFE_MSG_SetFcnCode - Sets the function code field in a message.
    CFE_MSG_GetFcnCode - Gets the function code field from a message.
    CFE_MSG_GetMsgTime - Gets the time field from a message.
    CFE_MSG_SetMsgTime - Sets the time field in a message.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Requester Info**
Alex Campbell GSFC

### PR DESCRIPTION
**Pull request by pavll**
**Describe the contribution**
- Fixes #1677 

**Contributor Info - All information REQUIRED for consideration of pull request**
Paul Oberosler, Individual

### CODE REVIEW

File: modules/msg/fsw/src/cfe_msg_ccsdsext.c [hunk 1]
```diff
@@ -90,7 +90,7 @@ CFE_Status_t CFE_MSG_GetEDSVersion(const CFE_MSG_Message_t *MsgPtr, CFE_MSG_EDSV
  *-----------------------------------------------------------------*/
 CFE_Status_t CFE_MSG_SetEDSVersion(CFE_MSG_Message_t *MsgPtr, CFE_MSG_EDSVersion_t Version)
 {
-    if (MsgPtr == NULL || (Version > (CFE_MSG_EDSVER_MASK >> CFE_MSG_EDSVER_SHIFT)))
+    if (MsgPtr == NULL || Version == NULL || (Version > (CFE_MSG_EDSVER_MASK >> CFE_MSG_EDSVER_SHIFT)))
```
Comment (skliper) [inline modules/msg/fsw/src/cfe_msg_ccsdsext.c:93]: Version can be 0...

File: modules/msg/fsw/src/cfe_msg_ccsdsext.c [hunk 2]
```diff
@@ -253,7 +253,7 @@ CFE_Status_t CFE_MSG_GetSubsystem(const CFE_MSG_Message_t *MsgPtr, CFE_MSG_Subsy
  *-----------------------------------------------------------------*/
 CFE_Status_t CFE_MSG_SetSubsystem(CFE_MSG_Message_t *MsgPtr, CFE_MSG_Subsystem_t Subsystem)
 {
-    if (MsgPtr == NULL || ((Subsystem & ~CFE_MSG_SUBSYS_MASK) != 0))
+    if (MsgPtr == NULL || Subsystem == NULL || ((Subsystem & ~CFE_MSG_SUBSYS_MASK) != 0))
```
Comment (skliper) [inline modules/msg/fsw/src/cfe_msg_ccsdsext.c:256]: Subsystem can be zero

File: modules/msg/fsw/src/cfe_msg_ccsdsext.c [hunk 3]
```diff
@@ -294,7 +294,7 @@ CFE_Status_t CFE_MSG_GetSystem(const CFE_MSG_Message_t *MsgPtr, CFE_MSG_System_t
  *-----------------------------------------------------------------*/
 CFE_Status_t CFE_MSG_SetSystem(CFE_MSG_Message_t *MsgPtr, CFE_MSG_System_t System)
 {
-    if (MsgPtr == NULL)
+    if (MsgPtr == NULL || System == NULL)
```
Comment (skliper) [inline modules/msg/fsw/src/cfe_msg_ccsdsext.c:297]: System can be 0

**Review by skliper — CHANGES_REQUESTED** [review]
Good work!  See requested changes... those values can all be zero (they aren't pointers).

**Comment by skliper** [comment]
Commit message typo - it should reference #1677.

File: modules/msg/fsw/src/cfe_msg_ccsdsext.c (hunk 1 above)
Comment (pavll) [reply modules/msg/fsw/src/cfe_msg_ccsdsext.c:93]: > Good work! See requested changes... those values can all be zero (they aren't pointers).

Thanks...

Sure, I'll change that back quickly.

**Comment by pavll** [comment]
> Commit message typo - it should reference #1677.

I also noticed, I will push it immediately

**Comment by pavll** [comment]
I have not yet managed to run a MemoryProfiler with the unit test to find out what checksum an empty message with id 2 has. Could someone help me with this? (for the ut of GenerateChecksum) 
[These are the lines](https://github.com/pavll/cFE/blob/04e1cf31923ee98543b637140b2f4c325271c357/modules/cfe_testcase/src/msg_api_test.c#L196-L208)

**Comment by skliper** [comment]
Duplicate of #1745

**Comment by skliper** [comment]
> I have not yet managed to run a MemoryProfiler with the unit test to find out what checksum an empty message with id 1 has. Could someone help me with this? (for the ut of GenerateChecksum)
> [These are the lines](https://github.com/pavll/cFE/blob/04e1cf31923ee98543b637140b2f4c325271c357/modules/cfe_testcase/src/msg_api_test.c#L196-L208)

See comments on #1745 - really GenerateChecksum just inserts the correct value into the message, so verification is just to run ValidateChecksum before and after and make sure it passes after.  There actually is no GetChecksum API anymore since we couldn't come up with a use-case from an application context.

**Pull request closed without merging**